nav2_regulated_pure_pursuit_controller/include/nav2_regulated_pure_pursuit_controller/regulated_pure_pursuit_controller.hpp

```diff
@@ -183,13 +189,20 @@ class RegulatedPurePursuitController : public nav2_core::Controller
     const double &, const double &);
 
   /**
-   * @brief Whether point is in collision
+   * @brief checks for collision at projected pose
    * @param x Pose of pose x
    * @param y Pose of pose y
+   * @param theta orientation of Yaw
+   * @param footprint_spec footprint of the robot unoriented
+   * @param traverse_unknown ignore UNKNOWN cost (255) as obstuctle
    * @return Whether in collision
    */
-  bool inCollision(const double & x, const double & y);
-
+  bool inCollision(
+    const double & x,
+    const double & y,
+    const double & theta,
+    const nav2_costmap_2d::Footprint & footprint_spec,
```

Comment:
Remove footprint / traverse unknown, those are both available in class variables that this function can access.